src/e84_geoai_common/llm/core/llm.py

```diff
@@ -54,30 +85,32 @@ def to_text_only(self) -> str:
         return " ".join(parts)
 
 
-class Tool(BaseModel):
-    """Defines a tool that the model may use."""
+class LLMTool(BaseModel):
+    """Definition of a tool that an LLM may use."""
 
     model_config = ConfigDict(strict=True, extra="forbid")
 
     name: str
     description: str
-    input_schema: dict[str, Any]
-
-    @classmethod
-    def from_function(cls, func: Callable[..., Any]) -> Self:
-        """Construct from a Python funtion."""
-        # This works because our tool class has the same fields as a Claude tool
-        schema = get_function_schema(func, format="claude")  # type: ignore[reportUnknownVariableType]
-        return cls.model_validate(schema)
+    input_model: type[BaseModel] | None = Field(
```

Comment:
I think I like this approach. It makes it less flexible since we can't create our own JSON schema and pass it it in but who would want to handcraft one anyway.